### DISCUSSION TITLE
Markdown files don't show strikethrough

### DISCUSSION THREAD

**Discussion by canle1404** [MeanderingProgrammer/render-markdown.nvim#256] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/d3ffd5d1-f56f-4311-9a65-4a59c8755719)
Hi there. Looks like this plugin supports strikethrough but it can't be enable with my config. I tried upgrading `render-markdown.nvim` and `tree-sitter` parsers: `markdown` and `markdown-inline` but it still didn't work.
This is my config
```lua
return {
  "MeanderingProgrammer/render-markdown.nvim",
  opts = {
	  checkbox = {
		  unchecked = { icon = "" },
		  checked = { icon = "", scope_highlight = "@markup.strikethrough" },
		  custom = {
			  doing = {
				  raw = "[_]",
				  rendered = "󰄮",
				  highlight = "RenderMarkdownDoing",
			  },
			  wontdo = {
				  raw = "[~]",
				  rendered = "󰅗",
				  highlight = "RenderMarkdownWontdo",
			  },
		  },
	  },
  },
  dependencies = { "nvim-treesitter/nvim-treesitter", "nvim-tree/nvim-web-devicons" },
}
```

**Comment by MeanderingProgrammer** [comment, marked as answer]
This isn't a feature of this plugin and instead It's up to your colorscheme to provide a reasonable value for the `@markup.strikethrough` highlight group.

If you run the command `:highlight @markup.strikethrough` you'll see the value that's currently set or get an error if one does not exist. This is the output for me: `@markup.strikethrough xxx cterm=strikethrough gui=strikethrough`.

If you don't want to change your color scheme you can also define the highlight group yourself at some point after loading your color scheme by calling:

```lua
vim.api.nvim_set_hl(0, '@markup.strikethrough', {
    strikethrough = true,
    force = true,
})
```

**Reply by egreb** [reply]
I can't get this to work. 

```lua
return {
  'folke/tokyonight.nvim',
  lazy = false,
  priority = 1000,
  opts = {},
  config = function()
    vim.cmd [[colorscheme tokyonight-storm]]
    -- enable strikethrough for markdown files
    vim.api.nvim_set_hl(0, '@markup.strikethrough', {
      strikethrough = true,
      force = true,
    })
  end,
}
```

**Reply by MeanderingProgrammer** [reply]
`folke/tokyonight.nvim` does this by default: https://github.com/folke/tokyonight.nvim/blob/main/lua/tokyonight/groups/treesitter.lua#L67

If `strikethrough` is not working then it's likely one of 2 things:

- You don't have `nvim-treesitter` highlights enabled

```lua
require('nvim-treesitter.configs').setup({
    highlight = { enable = true },
})
```

- Your terminal emulator does not properly support strikethrough text. If you run this command in your terminal `echo -e "\e[9mStrikethrough text\e[0m"` and it does not have a strikethrough then the issue is the terminal.

**Reply by egreb** [reply]
I am using ghostty and your echo example has a strike-through. 

This is my `nvim-treesitter` config:

```lua
return { -- Highlight, edit, and navigate code
  'nvim-treesitter/nvim-treesitter',
  build = ':TSUpdate',
  opts = {
    ensure_installed = { 'bash', 'diff', 'html', 'lua', 'luadoc', 'markdown', 'go', 'javascript', 'typescript' },
    auto_install = true,
    highlight = {
      enable = true,
    },
    indent = { enable = true, disable = { 'ruby' } },
  },
  config = function(_, opts)
    require('nvim-treesitter.install').prefer_git = true
    ---@diagnostic disable-next-line: missing-fields
    require('nvim-treesitter.configs').setup(opts)
  end,
}
```

Looks like it should work imo 😅

**Reply by MeanderingProgrammer** [reply]
Missing `markdown_inline` parser in `ensure_installed`.

The markdown parser is a bit unique in that blocks are handled separately from inline elements. The checkhealth for this plugin should show an error for the missing parser.

**Reply by egreb** [reply]
So, the ~~strike-through~~ works in `ghostty` but not in `wezterm`. I am using `wezterm` for my markdown notes so would be nice to get that working as well. 
You're echo to test strike-through works in wezterm though but not inside neovim

**Reply by MeanderingProgrammer** [reply]
Huh, just curious, why use different terminals for different file types?

What does `:checkhealth render-markdown` output?

**Reply by egreb** [reply]
Because I have a raycast script that starts wezterm with my notes open, I don't think I can do that with ghostty. 

```bash
open -a /opt/homebrew/bin/wezterm --args start --cwd ~/notes -- nvim -c "ZkNotes" .
```

Output from checkhealth:
```

==============================================================================
render-markdown: require("render-markdown.health").check()

render-markdown.nvim [version] ~
- OK plugin 7.8.9
- OK neovim >= 0.10

render-markdown.nvim [configuration] ~
- OK valid

render-markdown.nvim [nvim-treesitter] ~
- OK installed
- OK markdown: parser installed
- OK markdown: highlight enabled
- OK markdown_inline: parser installed
- OK markdown_inline: highlight enabled
- WARNING latex: parser not installed
  - ADVICE:
    - Disable LaTeX support to avoid this warning by setting { latex = { enabled = false } }

render-markdown.nvim [icons] ~
- OK using: mini.icons

render-markdown.nvim [executables] ~
- WARNING latex2text: not installed
  - ADVICE:
    - Disable LaTeX support to avoid this warning by setting { latex = { enabled = false } }

render-markdown.nvim [conflicts] ~
- OK headlines: not installed
- OK obsidian: not installed
```
